app/main.js

```diff
@@ -731,16 +735,21 @@ function getTrayMenu () {
       createPreferencesWindow()
     }
   }, {
-    type: 'separator'
+    type: 'separator',
+    visible: !settings.get('hardcoreMode')
   }, {
     label: i18next.t('main.quitStretchly'),
+    visible: !settings.get('hardcoreMode'),
     role: 'quit',
     click: function () {
       app.quit()
     }
   })
 
-  return Menu.buildFromTemplate(trayMenu)
+  // Hide the invisible items. Electron has a bug which ignores visible: false parameter for separator menuitems. Hide them here.
```

Comment:
can you point me to bug? I would like to subscribe to it

Reply:
https://github.com/electron/electron/issues/6906
https://github.com/electron/electron/issues/3494